airflow/configuration.py

```diff
@@ -330,13 +346,36 @@ def __init__(self, defaults, *args, **kwargs):
 
     def _validate(self):
         if self.get("core", "executor") != 'SequentialExecutor' \
-                and "sqlite" in self.get('core', 'sql_alchemy_conn'):
+                and "sqlite" in self.get_with_fallback('core', 'sql_alchemy_conn'):
             raise AirflowConfigException("error: cannot use sqlite with the {}".
                                                        format(self.get('core', 'executor')))
 
         self.is_validated = True
 
     def get(self, section, key, **kwargs):
+        return self._get(section, key, True)
+
+    def get_with_fallback(self, section, key, **kwargs):
+        ret_val = self._get(section, key, False)
```

Comment:
Setting a default on any of these would result in never ever getting to the command, which wouldn't be expected. I think the order should be 1-conf, 2-envvar, 3-command(if listed as a "commandable" param, 4-default)